src/WebJobs.Script/Config/ActiveHostConfigurationSource.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Script.Configuration
+{
+    public class ActiveHostConfigurationSource : IConfigurationSource
+    {
+        public ActiveHostConfigurationSource(IServiceProvider serviceProvider)
+        {
+            ServiceProvider = serviceProvider;
+        }
+
+        public IServiceProvider ServiceProvider { get; set; }
+
+        public IConfigurationProvider Build(IConfigurationBuilder builder)
+        {
+            return new ActiveHostConfigurationProvider(this);
+        }
+
+        private class ActiveHostConfigurationProvider : ConfigurationProvider
+        {
+            private readonly ActiveHostConfigurationSource _configurationSource;
+
+            public ActiveHostConfigurationProvider(ActiveHostConfigurationSource configurationSource)
+            {
+                _configurationSource = configurationSource ?? throw new ArgumentNullException(nameof(configurationSource));
+            }
+
+            public override bool TryGet(string key, out string value)
+            {
+                if (_configurationSource?.ServiceProvider?.GetService(typeof(IConfiguration)) is IConfigurationRoot activeHostConfiguration)
```

Comment:
this call will be simplified when you take the service provider, and the null check will also not be needed.